serde/src/java/org/apache/hadoop/hive/serde2/objectinspector/ObjectInspectorConverters.java

```diff
@@ -389,17 +442,32 @@ public Object convert(Object input) {
         return null;
       }
 
-      int minFields = Math.min(inputFields.size(), outputFields.size());
-      // Convert the fields
-      for (int f = 0; f < minFields; f++) {
-        Object inputFieldValue = inputOI.getStructFieldData(input, inputFields.get(f));
-        Object outputFieldValue = fieldConverters.get(f).convert(inputFieldValue);
-        outputOI.setStructFieldData(output, outputFields.get(f), outputFieldValue);
+      if (resolveByName) {
+        for (int f = 0; f < fieldConverters.size(); f++) {
+          StructFieldConverter converter = (StructFieldConverter) fieldConverters.get(f);
```

Comment:
This (done in both branches of the following if) can be moved here to make it clearer that we rely on outputFields & fieldConverters having the same ordering:

StructField out = outputFields.get(f);